src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -71,105 +78,135 @@ const NavHeader = (props) => {
+                paddingTop: insets.top,
+                paddingRight: dimen.normalizeDimen(20),
+                zIndex: 2,
+                position: 'absolute',
+                right: 0
+              })
+            }>
+            {!isFollow && isHeaderHide ? (
+              <ButtonFollow handleSetFollow={handleFollowTopic} />
+            ) : (
+              <TouchableOpacity onPress={onShareCommunity} style={styles.shareIconStyle}>
+                <ShareIconCircle color="black" width={32} height={32} />
+              </TouchableOpacity>
+            )}
+          </View>
         </View>
-      </View>
-      <Animated.View style={[styles.headerContainer]}>
-        {!isHeaderHide ? (
+        {!hasSearch && (
           <>
-            <Animated.View style={{opacity: opacityHeaderAnimation}}>
-              <FastImage
-                source={topicDetail?.icon_path ? {uri: topicDetail?.icon_path} : TopicDefaultIcon}
-                style={styles.image}
-              />
-            </Animated.View>
-            <View style={styles.containerAction}>
+            <View style={[styles.headerContainer]}>
               <Animated.View style={{opacity: opacityHeaderAnimation}}>
-                {isFollow ? (
-                  <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
-                ) : (
-                  <ButtonFollow handleSetFollow={handleFollowTopic} />
-                )}
+                <FastImage
+                  source={
+                    initialData?.channelPicutre
+                      ? {uri: initialData?.channelPicutre}
+                      : topicDetail?.icon_path
+                      ? {uri: topicDetail?.icon_path}
+                      : TopicDefaultIcon
+                  }
+                  style={styles.image}
+                />
               </Animated.View>
+              <View style={styles.containerAction}>
+                <Animated.View style={{opacity: opacityHeaderAnimation}}>
+                  {isFollow === undefined && isLoading ? (
+                    <Shimmer width={normalize(100)} height={normalize(36)} />
+                  ) : isFollow ? (
+                    <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
+                  ) : (
+                    <ButtonFollow handleSetFollow={handleFollowTopic} />
+                  )}
+                </Animated.View>
+              </View>
             </View>
+            <Animated.View
+              style={[
+                styles.domain(isHeaderHide),
+                {bottom: normalize(20), opacity: opacityHeaderAnimation}
+              ]}>
+              <TopicDomainHeader {...props} />
+            </Animated.View>
           </>
-        ) : null}
-      </Animated.View>
-      <View
-        onLayout={onDomainLayout}
-        style={[styles.domain(isHeaderHide), {bottom: getBottomPostition()}]}>
-        <TopicDomainHeader {...props} />
+        )}
       </View>
 
       {hasSearch && (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code seems to be well written and follows good practices. However, I noticed that the `opacityImage` prop is used directly in the styles (line 120, 143). It would be better to ensure that this value is within a valid range (0-1) before using it. This can prevent potential issues if an invalid value is passed as a prop.

```diff
- const {width: displayWidth} = useWindowDimensions();
+ const {width: displayWidth} = useWindowDimensions();
+ const safeOpacityImage = Math.max(0, Math.min(1, opacityImage));
```

Then replace all instances of `opacityImage` in the styles with `safeOpacityImage`.



<!-- This is an auto-generated comment by OSS CodeRabbit -->